added testing deeplinks (#9)
ScreaMy7/owasp-mastg: demos/android/MASVS-CODE/MASTG-DEMO-0063/MASTG-DEMO-0063.md

---
platform: android
title: Deep Link Intent Filter Missing android:autoVerify with semgrep
id: MASTG-DEMO-0063
code: [kotlin]
test: MASTG-TEST-0289
status: new
---

### Sample

The following is a sample `AndroidManifest.xml` snippet that defines a deep link intent filter without the `android:autoVerify="true"` attribute.

> {{ ../MASTG-DEMO-0062/AndroidManifest_reversed.xml }}

Reviewer: ```suggestion
{{ ../MASTG-DEMO-0062/AndroidManifest_reversed.xml # AndroidManifest_reversed.xml }}
```

### Steps

Let's run @MASTG-TOOL-0110 rules against the sample manifest.

{{ ../../../../rules/mastg-android-autoverify-missing.yml }}

{{ run.sh }}

### Observation

The rule has identified that the deep link intent filter is missing the `android:autoVerify="true"` attribute.

{{ output.txt }}

### Evaluation

> The test fails because the app does not enforce Android App Links verification. Without `android:autoVerify="true"`, malicious apps may intercept the app’s deep links, leading to phishing or hijacking attacks.

Reviewer: ```suggestion
The test fails because Android App Links verification is not enforced. Without `android:autoVerify="true"`, malicious apps can intercept deep links, enabling phishing or hijacking attacks.
```
